src/main/java/core/basesyntax/supplier/FigureSupplier.java

```diff
@@ -0,0 +1,39 @@
+package core.basesyntax.supplier;
+
+import core.basesyntax.Color;
+import core.basesyntax.figure.Circle;
+import core.basesyntax.figure.Figure;
+import core.basesyntax.figure.IsoscelesTrapezoid;
+import core.basesyntax.figure.Rectangle;
+import core.basesyntax.figure.RightTriangle;
+import core.basesyntax.figure.Square;
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int MAX_RANGE_BOUND = 10;
+    private final Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        Figure[] figures = {
```

Comment:
There is a lot of extra work in such a solution. We do not need to create all the figures at once in order to get one.
First, get the number by the number of figures, then do a switch case, and already do return new figure_type.
Also, to make it even better, you need to make separate private methods for creating each shape and call them in each case.

Reply:
Ok, I have done according your advices.